source/index.ts

```diff
@@ -110,7 +111,7 @@ function is(value: unknown): TypeName {
 		case 'string':
 			return 'string';
 		case 'number':
-			return 'number';
+			return Number.isNaN(value) ? 'NaN' : 'number';
```

Comment:
I think it would be clearer to do the `NaN` check on line 107.

Reply:
That would mean isNan is ran for everything but `null`, this way it is only ran for what we know `NaN` reports as typeof

Reply:
This optimization can likewise be done for object/null

Reply:
You are assuming `Number.isNaN` is not optimized. I feel like this is entering premature optimization territory. In practice, the performance difference is probably none or negligible.